furusiya/model/components/fighter.py

```diff
@@ -77,3 +77,5 @@ def die(self):
         death_function = self.death_function
         if death_function is not None:
             death_function(self.owner)
+        else:
```

Comment:
While the definition of `die` on `GameObject` is clever, it requires subclasses to call `super.die`. I think it makes more sense to think of `death_function` as an optional callback that, if present, gets called by `die`. I would recommend changing this.